src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -590,8 +602,6 @@ private void CreateGrainInstance(string grainTypeName, ActivationData data, stri
 
             activations.IncrementGrainCounter(grainClassName);
 
-            data.GrainInstance.Data = data;
```

Comment:
It is at line 575 now - trying to collect the two runtime injections into the grain instance into one place.
https://github.com/jthelin/orleans/commit/db5c3ead5b6ed342dff058f6f71cb1f3a2a844c2#diff-bcfe9efaf8ce10fd652fca0dd5a4d116R575

`grain` and `data.GrainInstance` are the same thing, but previous code doesn't make that very clear.